ext/openssl/ossl_x509store.c

```diff
@@ -329,7 +328,23 @@ ossl_x509store_set_trust(VALUE self, VALUE trust)
 static VALUE
 ossl_x509store_set_time(VALUE self, VALUE time)
 {
-    rb_iv_set(self, "@time", time);
+    X509_STORE *store;
+    X509_VERIFY_PARAM *param;
+
+    GetX509Store(self, store);
+#if HAVE_X509_STORE_GET0_PARAM
+    param = X509_STORE_get0_param(store);
+#else
+    param = X509_VERIFY_PARAM_new();
+#endif
+    X509_VERIFY_PARAM_set_time(param, NUM2LONG(rb_Integer(time)));
+#if !HAVE_X509_STORE_GET0_PARAM
+    if (!X509_STORE_set1_param(store, param))
+    {
+        X509_VERIFY_PARAM_free(param);
```

Comment:
JFYI; you may want to unconditionally call `X509_VERIFY_PARAM_free`, since `X509_STORE_set1_param` appears to perform a state copy rather than transfering ownership of the `param` that gets passed in. Consequently you may end up leaking memory in the "happy" case here.

(OpenSSL's docs are unclear on this, unfortunately -- this is from me tracing OpenSSL's `x509_lu.c` by hand.)

Reply:
Yes, in OpenSSL's convension `set0` is used for functions that transfer ownership, and `set1` is used for those that copy contents or increment a reference counter.